site/_js/firebase.js

```
@@ -0,0 +1,118 @@
+      //   messagingSenderId: "802581884830",
+      //   appId: "1:802581884830:web:54daed54011bd73be6f08f"
+      // }
+    }
+
+    //Initialize firebase app
+    this.firebase = initializeApp(this.config);
+    // this.appCheck = initializeAppCheck(this.firebase, {
+    //   provider: new ReCaptchaV3Provider('6LdHHNweAAAAAA6jtgdZoZ6AFHC2_R4bSnAKY6Fp'),
+    //   isTokenAutoRefreshEnabled: true
+    // });
+    //this.appCheck.activate();
+
+    //Connect to Database
+    this.db = getDatabase(this.firebase);
+    //Getting chrome100 database
+    this.cardsRef = ref(this.db, 'chrome100');
+    //this.cardsRef = ref(this.db,'oct-sip-test2-default-rtdb');
+
+    this.updateLikesCount = this.updateLikesCount.bind(this);
+    this.initialize();
+    this.readData(this.cardsRef);
+  }
+
+  initialize() {
+    const buttons = document.querySelectorAll('.heart-logo');
+    const self = this;
+    for (const button of buttons) {
+      button?.addEventListener('click', function () {
+        const state = this.classList.toggle('like');
+        const {id, year} = this.dataset;
+        const storedCard = localStorage.getItem(id);
+        if (storedCard !== null) {
+          self.checkExisis(year, JSON.parse(storedCard));
+        } else {
+          self.updateLikesCount(year, id, state);
+        }
+      });
+    }
+  }
+
+  readData(cards) {
+    onValue(cards, snapshot => {
+      const data = snapshot.val();
+      console.log(data);
+      for (const year in data) {
+        this.updateLikeCount(year, data[year]);
+      }
+    });
+  }
+
+  // Update the likes value
+  updateLikeCount(year, data) {
+    const section = document.getElementById(`section${year}`);
+    const childrens = section?.querySelectorAll('.like-count');
+    childrens.forEach((child, idx) => {
+      child.innerHTML = `${data[idx].likes}`;
+    });
+  }
+
+  // check the card exist on local storage or not
+  checkExisis(year, key) {
+    runTransaction(this.cardsRef, post => {
```

I don't understand why this is being wrapped in [`runTransaction()`](https://firebase.google.com/docs/reference/js/database#runtransaction). Could you please explain? If you do need to use `runTransaction()` for some reason, why is the `Promise<TransactionResult>` return value being silently dropped?

Though honestly, I don't think this project is supposed to be using the Realtime Database anyway (see earlier comment), so the answer might be moot.

---

Based on the Real time Database implementation, we have used this function.
Reference: https://firebase.google.com/docs/database/web/read-and-write